### PR TITLE
Centralize Warning Handling

### PR DESCRIPTION
**Pull request by Kashif-S**
Jira: GLT-2389

### CODE REVIEW

File: miso-web/src/main/webapp/scripts/warning.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+WarningTarget = {};
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/warning.js]: BLOCKER Add the "let", "const" or "var" keyword to this declaration of "WarningTarget" to make it explicit. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)

File: miso-web/src/main/webapp/scripts/warning_completion.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * *********************************************************************
+ */
+
+//TODO: add consent revoked errors
+
+WarningTarget.completion = {
+    tableWarnings: function(data, type, full){
+      var warnings = [];
+      pool = full.pool;
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/warning_completion.js]: BLOCKER Add the "let", "const" or "var" keyword to this declaration of "pool" to make it explicit. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2703)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java [hunk 3]
```diff
@@ -67,6 +67,7 @@
 import uk.ac.bbsrc.tgac.miso.core.data.Platform;
 import uk.ac.bbsrc.tgac.miso.core.data.Pool;
 import uk.ac.bbsrc.tgac.miso.core.data.Sample;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryDilution;
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java]: MINOR Remove this unused import 'uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryDilution'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by sqwci** [comment]
SonarQube analysis reported 13 issues
* ![BLOCKER][BLOCKER] 3 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EditPoolController.java#L314](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L314): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MINOR][MINOR] [EditPoolController.java#L33](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L33): Remove this unused import 'java.util.Set'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L59](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L59): Remove this unused import 'com.fasterxml.jackson.core.JsonProcessingException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L142](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L142): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L170](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L170): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L175](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L175): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L253](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L253): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L269](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L269): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L318](https://github.com/TGAC/miso-lims/blob/664176a1b3c12eeedd6d916f59bccf14a2ef8091/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L318): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-web/src/main/webapp/scripts/warning_completion.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+//TODO: add consent revoked errors
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/warning_completion.js]: Is there another ticket for this TODO?

File: miso-web/src/main/webapp/scripts/warning_experiment.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+//TODO: add consent revoked errors
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/warning_experiment.js]: Same TODO

File: miso-web/src/main/webapp/scripts/warning_partition.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+//TODO: add consent revoked errors
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/warning_partition.js]: Same TODO

File: miso-web/src/main/webapp/scripts/warning_completion.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+WarningTarget.completion = {
+    tableWarnings: function(data, type, full){
+      var warnings = [];
+      pool = full.pool;
+      if(pool.duplicateIndices){
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/warning_completion.js]: A lot of this code (if data.possibilityX warn x; if data.possibiltyY warn y; ...) is duplicated across the various WarningTarget files. Can it be abstracted into a method?

File: miso-web/src/main/webapp/scripts/warning_pool.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+//TODO: add consent revoked errors
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/warning_pool.js]: Same TODO

**Comment by sqwci** [comment]
SonarQube analysis reported 10 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EditPoolController.java#L314](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L314): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MINOR][MINOR] [EditPoolController.java#L33](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L33): Remove this unused import 'java.util.Set'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L59](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L59): Remove this unused import 'com.fasterxml.jackson.core.JsonProcessingException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L142](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L142): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L170](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L170): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L175](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L175): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L253](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L253): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L269](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L269): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L318](https://github.com/TGAC/miso-lims/blob/2e31f084ca63f7b5d3b7a77cc72ec4955e0a02d7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L318): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 10 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EditPoolController.java#L314](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L314): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MINOR][MINOR] [EditPoolController.java#L33](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L33): Remove this unused import 'java.util.Set'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L59](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L59): Remove this unused import 'com.fasterxml.jackson.core.JsonProcessingException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L142](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L142): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L170](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L170): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L175](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L175): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L253](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L253): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L269](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L269): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L318](https://github.com/TGAC/miso-lims/blob/41c358efae6f1e0ed2bd0f2c5b0ba47df51eade1/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L318): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 10 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EditPoolController.java#L314](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L314): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MINOR][MINOR] [EditPoolController.java#L33](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L33): Remove this unused import 'java.util.Set'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L59](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L59): Remove this unused import 'com.fasterxml.jackson.core.JsonProcessingException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L142](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L142): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L170](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L170): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L175](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L175): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L253](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L253): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L269](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L269): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L318](https://github.com/TGAC/miso-lims/blob/4832ad8bb45cfdd48b9d942d9384e44b3d45ae3d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L318): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 10 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EditPoolController.java#L314](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L314): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MINOR][MINOR] [EditPoolController.java#L33](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L33): Remove this unused import 'java.util.Set'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L59](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L59): Remove this unused import 'com.fasterxml.jackson.core.JsonProcessingException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [EditPoolController.java#L142](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L142): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L170](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L170): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L175](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L175): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L253](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L253): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L269](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L269): Replace "@RequestMapping(method = RequestMethod.POST)" with "@PostMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)
1. ![MINOR][MINOR] [EditPoolController.java#L318](https://github.com/TGAC/miso-lims/blob/fd177f89751e7cc45ced6f6f823c790f1ad690ef/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java#L318): Replace "@RequestMapping(method = RequestMethod.GET)" with "@GetMapping" [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4488)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**